### PR TITLE
Fixes the fix for the sceditor xd

### PR DESCRIPTION
**Pull request by DiegoAndresCortes**
Fixes issues caused by #5426 
and extra fixes for #5370 ??

😋

### CODE REVIEW

**Comment by DiegoAndresCortes** [comment]
I'm still testing this btw but forgot to mark as draft

**Comment by DiegoAndresCortes** [comment]
I think it's ready if you want to test it

**Comment by DiegoAndresCortes** [comment]
So supposedly it shouldn't shrink, but allowing the editor to shrink below the set height is the ideal solution here.

Obviously the main issue #5370 it's solved, however when resizing the box it will set this "custom" height to the editor which causes this again. **BUT** the main issue is not dealing with this when the user is not playing around with the size of the editor (the default size), which should now be fixed without causing any _other_ problems.

Well, also forgot to mention that _it would only invoke_ the issue when the user **resizes the browser window** because obviously that's the only way to increase the height of the bbc/emoticons in real time.

So naturally, this is fixed imo.

Thanks for reading then, this was indeed a tricky one.

File: Themes/default/css/jquery.sceditor.css [hunk 1]
```diff
@@ -197,24 +198,24 @@ div.sceditor-dropdown div {
...
 	resize: none;
 	background: #fff;
-	display: block;
+	display: flex;
+	flex-basis: 175px;
+	flex-grow: 1;
 	min-width: 100%;
 	max-width: 100%;
-	height: auto !important;
+	height: 175px;
```
Comment (live627) [inline Themes/default/css/jquery.sceditor.css:216]: Now what would happen if a textedit control/mod specifies a longer height? Should this then be  a custom data attribute?

Comment (DiegoAndresCortes) [reply Themes/default/css/jquery.sceditor.css:216]: Can your JavaScript code target this using height instead of minHeight?

Comment (DiegoAndresCortes) [reply Themes/default/css/jquery.sceditor.css:216]: Is there an example of this behavior also?? To look into it

**Comment by DiegoAndresCortes** [comment]
I believe this one fixes the issue @live627

File: Themes/default/scripts/jquery.sceditor.smf.js [hunk 2]
```diff
@@ -932,4 +932,4 @@ sceditor.formats.bbcode.set(
 		},
 		html: '<div class="videocontainer"><div><iframe frameborder="0" src="https://www.youtube.com/embed/{0}?wmode=opaque" data-youtube-id="{0}" allowfullscreen></iframe></div></div>'
 	}
-);
+);
```
Comment (XinYenFon) [inline Themes/default/scripts/jquery.sceditor.smf.js:935]: my sweet innocent file endings :|

**Comment by sbulen** [comment]
I definitely like this behavior better, it feels more solid.  Three odd things I've seen...  

(1) You can size the window _smaller_ than available space, & create some funky behavior.  

E.g., I can shrink from here:
image

To here.  Note the text is completely gone, & the 'attachments' line can be scrunched/hidden as well:
image

(2) The other thing is that the MORE window is massive and obscures everything (100% browser zoom):
image

(3) Resize the browser window itself (to 150%) & the MORE window appears to be transparent:
image


Are these fixable?

**Comment by DiegoAndresCortes** [comment]
1 is tricky I commented on this behavior above, it seems to be the best solution but I’ll play around with it a bit more and see what can be improved.

2n3 I didn’t test those, I’ll make sure to have a look.

Also I don’t have internet atm so a few days at least.

**Comment by SychO9** [comment]
@sbulen why is your smiley box like that ? it should be one line,

the best way to fix that issue is for use to force the smiley box to a fixed height and avoid more lines of smileys, that's why we have the "More" button in the first place, because we only show a few of them inline and everything in the modal box.

**Comment by XinYenFon** [comment]
1) Resizing issue seems from our side... 
2) That's default behavior from SCEditor side, I strongly suggest it to keep it as is...
3) `.sceditor-container, .sceditor-container div, div.sceditor-dropdown, div.sceditor-dropdown div` has low z-index... increasing it to 5 (or removing it) solves the issue (for now)***

*** The default behavior for "More" is just to extend the list unlike our approach where you create extra-pop... 

@SychO9 You can enable custom smiley ordering and create massive pain to developers...

**Comment by SychO9** [comment]
> @SychO9 You can enable custom smiley ordering and create massive pain to developers...

Oh boi...

**Comment by sbulen** [comment]
I've seen users with dozens of them...

**Comment by XinYenFon** [comment]
Well, since (by design) SCEditor has extend (while SMF has pop), the inline elements needs to become block elements this will solve some of the issues...

For next version we can discuss how to change this feature (if required or not).

**Comment by live627** [comment]
> That's default behavior from SCEditor side, I strongly suggest it to keep it as is...

No, it's custom code that I helped write.

**Comment by XinYenFon** [comment]
> > That's default behavior from SCEditor side, I strongly suggest it to keep it as is...
> 
> No, it's custom code that I helped write.

hmm... why may i ask ? Default behavior seems good enough. I don't recall the reasoning behind moving the "more" part into a popup container.

**Comment by DiegoAndresCortes** [comment]
So the z-index fix is the only thing we want to do for now?

**Comment by XinYenFon** [comment]
yes (?) unless you guys think its good idea to revert "More" func. back into the original one

**Comment by sbulen** [comment]
Is there nothing that can be done about the size of the more box?

**Comment by XinYenFon** [comment]
Its size already defined inside CSS, we can simply resize it and be done with it

**Comment by SychO9** [comment]
@DiegoAndresCortes can you please push a fix for the height of the "more" smileybox ? I think you just have to remove the `height: 50%` from `div.sceditor-smileyPopup` and add `max-height: 50%` & `overflow-y: auto` to `div.sceditor-smileyPopup, div.sceditor-dropdown`

I think this should go into RC3 because of how obvious the bug is :/
repost

**Comment by DiegoAndresCortes** [comment]
Why is this box "fixed" instead of "absolute". Is that the expected behavior we want?

**Comment by SychO9** [comment]
> Why is this box "fixed" instead of "absolute". Is that the expected behavior we want?

yea, I think if it was `absolute` it would just be floating at the top of the page, so when you click more you actually wouldn't see it until you scroll all the way up. that's why `fixed` is more appropriate.

**Comment by MissAllSunday** [comment]
Hi @DiegoAndresCortes whats the status on this one? does testing is still required or do you feel its ready to go?

**Comment by DiegoAndresCortes** [comment]
Should be ready

**Review by MissAllSunday — APPROVED** [review]

**Pull request merged**